libraries/contracts/LibOrderData.sol

```diff
@@ -50,21 +50,45 @@ library LibOrderData {
         return payout;
     }
 
-    function parseOriginFeeData(uint data) internal pure returns(LibPart.Part[] memory) {
-        LibPart.Part[] memory originFee = new LibPart.Part[](1);
-        originFee[0].account = payable(address(data));
-        originFee[0].value = uint96(data >> 160);
+    function parseOriginFeeData(uint dataFirst, uint dataSecond) internal pure returns(LibPart.Part[] memory) {
+        LibPart.Part[] memory originFee;
+
+        if (dataFirst > 0 && dataSecond > 0){
+            originFee = new LibPart.Part[](2);
+
+            originFee[0].account = payable(address(dataFirst));
```

Comment:
How do you think 
Block code Line 59,60 looks like duplicate code (the same in line 62,62; 69,70; 76,77)
Maybe better organize  Line 59,60 into small internal function and use it
Seems it don`t add large gas consumption?

Reply:
great idea, fixed